### PR TITLE
Fix-Keep date picker open when selecting time, month, year in Table widget filters

### LINKED ISSUE #4277: [Bug] Date filter in the table doesn't let you type a time
### Steps to reproduce the behaviour:
1. Choose a column type as date
2. Try to filter by date, but notice that you can't type anything in the time input.
3. Just clicking the time input closes the popup
https://www.loom.com/share/763d5181b9f941c4bfc147ae40c8d2b4

### PR DESCRIPTION
**Pull request by vicky-primathon**
## Description
Keep date picker open when selecting time, month, year in Table widget filters

Fixes #4277 

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:white_circle: Total coverage has not changed</summary>

    // Code coverage diff between base branch:release and head branch: fix/table-date-filters 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | app/client/src/components/designSystems/blueprint/DatePickerComponent2.tsx | 23.81 **(0.28)** | 2.99 **(0)** | 5 **(0)** | 23.75 **(0.29)**</details>

### CODE REVIEW

File: app/client/src/widgets/TableWidget/derived.js [hunk 1]
```diff
@@ -364,16 +364,16 @@ export default {
...
-        return !moment(a).isSame(moment(b), "d");
+        return !moment(a).isSame(moment(b), "minute");
       },
       isAfter: (a, b) => {
-        return !moment(a).isAfter(moment(b), "d");
+        return moment(a).isAfter(moment(b), "minute");
       },
       isBefore: (a, b) => {
-        return !moment(a).isBefore(moment(b), "d");
+        return moment(a).isBefore(moment(b), "minute");
```
Comment (SatishGandham) [inline app/client/src/widgets/TableWidget/derived.js:376]: What is the purpose of minute, what does that mean?

Comment (vicky-primathon) [reply app/client/src/widgets/TableWidget/derived.js:376]: compare date along with hour and minutes.

**Comment by YogeshJayaseelan** [comment]
@vicky-primathon Now im able to set day, time, month and year working as expected

**Review by SatishGandham — APPROVED** [review]

**Comment by somangshu** [comment]
@vicky-primathon please clear all conflicts here

**Review by somangshu — APPROVED** [review]

**Review by SatishGandham — APPROVED** [review]

**Pull request merged**